converter/src/main/java/org/dswarm/converter/morph/MorphScriptBuilder.java

```diff
@@ -344,66 +348,56 @@ private void createLookupTable(final Element maps, final Component transformatio
-
-								case "in":
-								case "map":
-
-									map.setAttribute("name", parameterMapping.getValue());
-									break;
+						if (parameterMapping.getKey().equals(LOOKUP_MAP_DEFINITION) && parameterMapping.getValue() != null) {
 
-								case LOOKUP_MAP_DEFINITION:
+							switch (component.getFunction().getName()) {
 
+								case METAMORPH_FUNCTION_WHITELIST:
+								case METAMORPH_FUNCTION_BLACKLIST:
 
-									if (component.getFunction().getName().equals(METAMORPH_FUNCTION_WHITELIST)
-											|| component.getFunction().getName().equals(METAMORPH_FUNCTION_BLACKLIST)) {
+									try {
 
-										try {
+										final List<String> lookupList = DMPPersistenceUtil.getJSONObjectMapper().readValue(parameterMapping.getValue(),
+												new TypeReference<List<String>>() {});
 
-											final List<String> lookupList = DMPPersistenceUtil.getJSONObjectMapper().readValue(parameterMapping.getValue(),
-													new TypeReference<List<String>>() {});
+										for (final String lookupEntry : lookupList) {
 
-											for (final String lookupEntry : lookupList) {
-
-												final Element lookup = doc.createElement(METAMORPH_ELEMENT_MAP_ENTRY);
-												lookup.setAttribute("name", lookupEntry);
-												map.appendChild(lookup);
-											}
-										} catch (final IOException e) {
-
-											MorphScriptBuilder.LOG.debug("lookup map as JSON string in parameter mappings could not convert to a list" + e);
+											final Element lookup = doc.createElement(METAMORPH_ELEMENT_MAP_ENTRY);
+											lookup.setAttribute("name", lookupEntry);
+											map.appendChild(lookup);
 										}
+									} catch (final IOException e) {
+
+										MorphScriptBuilder.LOG.debug("lookup map as JSON string in parameter mappings could not convert to a list" + e);
+									}
+									break;
 
-									} else {
+								case METAMORPH_FUNCTION_LOOKUP:
+								case METAMORPH_FUNCTION_SETREPLACE:
 
-										try {
+									try {
 //
-											final Map<String, String> lookupEntrys = DMPPersistenceUtil.getJSONObjectMapper().readValue(parameterMapping.getValue(),
-													new TypeReference<HashMap<String, String>>() {});
+										final Map<String, String> lookupEntrys = DMPPersistenceUtil.getJSONObjectMapper().readValue(parameterMapping.getValue(),
+												new TypeReference<HashMap<String, String>>() {});
 
-											for (final Entry<String, String> lookupEntry : lookupEntrys.entrySet()) {
+										for (final Entry<String, String> lookupEntry : lookupEntrys.entrySet()) {
 
-												final Element lookup = doc.createElement(METAMORPH_ELEMENT_MAP_ENTRY);
-												lookup.setAttribute("name", lookupEntry.getKey());
-												lookup.setAttribute("value", lookupEntry.getValue());
-												map.appendChild(lookup);
+											final Element lookup = doc.createElement(METAMORPH_ELEMENT_MAP_ENTRY);
+											lookup.setAttribute("name", lookupEntry.getKey());
```

Comment:
"name" and "value" could also be a constant here, or? (i.e. private static final ...)